app/controllers/users_controller.rb

```diff
@@ -22,9 +40,40 @@ def create
     end
   end
 
+  def update
+    if @user.update(user_params)
+      flash[:success] = "Profile updated"
+      redirect_to @user
+    else
+      render "edit"
+    end
+  end
+
+  def edit; end
+
   private
+  def load_user
+    @user = User.find_by(id: params[:id])
```

Comment:
thêm code xử lý nil nhé 
`return if @user`
`flash[:warning] = "User not found"`
`redirect_to root_path`